### PR TITLE
Adds ISF integration for the editor.

### PR DESCRIPTION
**Pull request by dbatiste**
This PR includes:

* tinyMCE plugin (button for launching dialog and processing/inserting HTML)
* ISF styles for the content/design view of the editor
* ISF dialog component (wraps calls to LMS dialog)

The extent to which we need to continue supporting some of these classes and scenarios remains an open question.  The processing logic is basically copied from existing and modified for ES6 linting.  We'll be taking a broader look at media insertion in the near future so probably not worthwhile introducing any potentially disruptive changes here.

### CODE REVIEW

File: components/isf.js [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+		border: 1px solid #006fbf;
+	}
+`.cssText;
+
+tinymce.PluginManager.add('d2l-isf', function(editor) {
+
+	// bail if no LMS context
+	if (!D2L.LP) return;
+
+	const event = new CustomEvent('d2l-request-instance', {
```
Comment (dbatiste) [inline components/isf.js:33]: I may create a reusable helper in core for this.

File: components/isf.js [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+			dialog.opened = true;
+			dialog.addEventListener('d2l-htmleditor-isf-dialog-close', (e) => {
+				const html = e.detail.html;
+				if (html) editor.execCommand('mceInsertContent', false, html);
+			}, { once: true });
+
+		}
+	});
+
+	const regExp = {
```
Comment (dbatiste) [inline components/isf.js:62]: This processing logic is basically copied and modified to pass ES6 linting as described.

File: components/isf.js [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+import { css, LitElement } from 'lit-element/lit-element.js';
+import { getComposedActiveElement } from '@brightspace-ui/core/helpers/focus.js';
+import { icons } from '../icons.js';
+import { RequesterMixin } from '@brightspace-ui/core/mixins/provider-mixin.js';
+
+// TODO: localize the tooltip
+
+export const isfStyles = css`
+	/* stylelint-disable-next-line selector-class-pattern */
+	.disf_default, .disf_flash, .disf_shockwave, .disf_quicktime, .disf_windowsmedia, .disf_realmedia {
```
Comment (grant-cleary) [inline components/isf.js:11]: Haha Flash and Quicktime...

Comment (dbatiste) [reply components/isf.js:11]: Gonna review this... might be able to remove some of this since I think a generic image is displayed.

**Review by grant-cleary — APPROVED** [review]

**Pull request merged**